src/mono/wasm/runtime-test.js

```diff
@@ -224,44 +227,49 @@ var Module = {
-			config.deploy_prefix,
-			config.enable_debugging,
-			config.assembly_list,
-			function () {
-				App.init ();
-			},
-			function (asset)
-			{
-			  // for testing purposes add BCL assets to VFS until we special case File.Open
-			  // to identify when an assembly from the BCL is being open and resolve it correctly.
-			  var content = new Uint8Array (read (asset, 'binary'));
-			  var path = asset.substr(config.deploy_prefix.length);
-			  writeContentToFile(content, path);
-
-			  if (typeof window != 'undefined') {
-				return fetch (asset, { credentials: 'same-origin' });
-			  } else {
-				// The default mono_load_runtime_and_bcl defaults to using
-				// fetch to load the assets.  It also provides a way to set a 
-				// fetch promise callback.
-				// Here we wrap the file read in a promise and fake a fetch response
-				// structure.
-				return new Promise((resolve, reject) => {
-						var response = { ok: true, url: asset,
-							arrayBuffer: function() {
-								return new Promise((resolve2, reject2) => {
-									resolve2(content);
-							}
-						)}
-					}
-				   resolve(response)
-				 })
-			  }
-			}
-		);
-	},
+
+        config.loaded_cb = function () {
+            App.init ();
+        };
+        config.fetch_file_cb = function (asset) {
+            // for testing purposes add BCL assets to VFS until we special case File.Open
+            // to identify when an assembly from the BCL is being open and resolve it correctly.
+            var content = new Uint8Array (read (asset, 'binary'));
+            var path = asset.substr(config.deploy_prefix.length);
+            writeContentToFile(content, path);
+
+            if (typeof window != 'undefined') {
+            	return fetch (asset, { credentials: 'same-origin' });
+            } else {
+                // The default mono_load_runtime_and_bcl defaults to using
+                // fetch to load the assets.  It also provides a way to set a
+                // fetch promise callback.
+                // Here we wrap the file read in a promise and fake a fetch response
+                // structure.
+	            return new Promise ((resolve, reject) => {
+	                var bytes = null, error = null;
+	                try {
+	                    bytes = read (asset, 'binary');
+	                } catch (exc) {
+	                    error = exc;
+	                }
+	                var response = { ok: (bytes && !error), url: asset,
+	                    arrayBuffer: function () {
+	                        return new Promise ((resolve2, reject2) => {
+	                            if (error)
+	                                reject2 (error);
+	                            else
+	                                resolve2 (new Uint8Array (bytes));
+	                    }
+	                )}
+	                }
+	                resolve (response);
+	            })
+            }
+        };
+
+        MONO.mono_load_runtime_and_bcl_args (config);
+    },
```

Comment:
it looks like we're using tabs instead of spaces in the rest of this file so we should be consistent.